test/TNBase.External.Tests/DataExport/CsvExportServiceTests.cs

```diff
@@ -0,0 +1,176 @@
+            listener.Telephone = "01234567890";
+            listener.Town = "London";
+            listener.Wallet = 0;
+            listener.Stock = 3;
+
+            return listener;
+        }
+
+        private Listener CreateListener2()
+        {
+            Listener listener = new Listener();
+
+            listener.Title = "Mrs";
+            listener.Forename = "Sarah";
+            listener.Surname = "Turner";
+            listener.Addr1 = "3 High Street";
+            listener.Addr2 = "near, here";
+            listener.Postcode = "WS2 4TF";
+            listener.County = "Shropshire";
+            listener.InOutRecords = new InOutRecords();
+            listener.MemStickPlayer = false;
+            listener.LastIn = DateTime.ParseExact("21/07/2020", DateHelpers.DEFAULT_DATE_FORMAT, null);
+            listener.LastOut = null;
+            listener.Joined = DateTime.ParseExact("01/12/2019", DateHelpers.DEFAULT_DATE_FORMAT, null);
+            listener.Magazine = false;
+            listener.Status = ListenerStates.ACTIVE;
+            listener.Telephone = "124";
+            listener.Town = "Walsall";
+            listener.Wallet = 1;
+            listener.Stock = 2;
+
+            return listener;
+        }
+
+        [Fact]
+        public void ExportListeners_ShouldHeaderOnly_WhenNoRecords()
+        {
+            using var context = new TNBaseContext("Data Source=:memory:");
+            context.UpdateDatabase();
+            var service = new CsvExportService();
+
+            string result = service.ExportListeners(context.Listeners.Local.ToList());
+
+            Assert.Equal(Header + Environment.NewLine, result);
+        }
+
+        [Fact]
+        public void ExportListeners_ShouldTwoLines_WhenOneRecord()
+        {
+            using var context = new TNBaseContext("Data Source=:memory:");
+            context.Listeners.Add(CreateListener1());
+            context.UpdateDatabase();
+            var service = new CsvExportService();
+
+            string result = service.ExportListeners(context.Listeners.Local.ToList());
+
+            Assert.Equal(3, result.Split(Environment.NewLine).Length);
+            Assert.Equal(Header, result.Split(Environment.NewLine)[0]);
+            Assert.NotEqual(Header, result.Split(Environment.NewLine)[1]);
+        }
+
+        [Fact]
+        public void ExportListeners_CorrectLine_WhenOneRecord()
+        {
+            using var context = new TNBaseContext("Data Source=:memory:");
+            context.Listeners.Add(CreateListener1());
+            context.UpdateDatabase();
+            var service = new CsvExportService();
+
+            string result = service.ExportListeners(context.Listeners.Local.ToList());
+
+            Assert.Equal("Ted,Baker,0,Mr,29 Baker Street,,London,Londonshire,N193HH,False,True,01234567890,01/01/2015,1,1,ACTIVE,,,3,0,21/02/2015,,,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0", result.Split(Environment.NewLine)[1]);
```

Comment:
Something to think about but every field added, removed, or updated would cause all these tests to fail. Ideally, for any change of behavior, only tests related to it should fail. At the same time, no test should fail when adding new behavior, i.e. adding a new field. I don't have a good solution right off the bat and certainly not asking you to change tests. Please don't. It's just something to keep in mind for the future.